encoding/dot/decode_test.go

```diff
@@ -0,0 +1,210 @@
+package dot_test
```

Comment:
The reason dot_test was used, is because of collisions with the Node type, and to serve as an example of the package usage. However, none of those reasons truly justify it, so I'll rename the Node type to something else, and potentially add an example to `dot_example.go`.